sdk/src/gateway/client.ts

```diff
@@ -115,12 +127,47 @@ export class GatewayApiClient {
         return Object.values(Chain);
     }
 
-    /**
-     * Get a quote from the Gateway API for swapping or staking BTC.
-     *
-     * @param params The parameters for the quote.
-     */
+    async getQuote(params: {
+        type: 'onramp';
+        params: Optional<
+            GatewayQuoteParams,
+            'amount' | 'fromChain' | 'fromToken' | 'fromUserAddress' | 'toUserAddress'
+        >;
+    }): Promise<GatewayQuote & GatewayTokensInfo>;
+
+    async getQuote(params: {
+        type: 'offramp';
+        params: {
+            tokenAddress: Address;
+            amount: number;
+        };
+    }): Promise<OfframpQuote>;
+
     async getQuote(
+        params:
+            | {
+                  type: 'onramp';
+                  params: Optional<
+                      GatewayQuoteParams,
+                      'amount' | 'fromChain' | 'fromToken' | 'fromUserAddress' | 'toUserAddress'
+                  >;
+              }
+            | {
+                  type: 'offramp';
+                  params: {
+                      tokenAddress: Address;
+                      amount: number;
+                  };
+              }
```

Comment:
very nit: Since the type definitions for `getQuote` are duplicated above, consider extracting a shared type for both `onramp` and `offramp` quotes.